### DISCUSSION TITLE
Merge different consecutive action types

### DISCUSSION THREAD

**Discussion by stijnvanbossuyt** [nilsmehlhorn/ngrx-wieder#82] [Ideas]
Hello,

I have a scenario where a have multiple actions mutating the state consecutively. Due to reusability of these actions they have been split up. 

I have the following 2 scenario's: 

Action: setQuestion
Action: setProperty

Action: setSubQuestion
Action: setProperty

To undo this I have to run the Undo action twice. Is it possible to define a merge strategy where I can define these actions as one state change?

For example something like this:

`mergeStrategy: [
   ['setQuestion', 'setProperty'],
   ['setSubQuestion', 'setProperty'],
]`

Thanks

**Comment by nilsmehlhorn** [comment]
Hi @stijnvanbossuyt,

I see your point and I've also been thinking that the current configuration API could be more flexible. E.g. the merge predicates defined currently by `mergeActionTypes` and  `mergeRules` could be consolidated into a single predicate that runs on any consecutive actions to merge their patches instead of being limited to the same action types.

Taking this thought further, this could also apply to `allowedActionTypes` which could be a singular predicate instead of a list of static actions types.

Would you be able to contribute to such an API change?

You might also be able to workaround the current limitation by adding an effect to your application which emits another undo/redo action depending on the last action that was undone/redone. You can get this information via the `selectHistory` selector`.
